src/common/schedulers/slurm_commands.py

```diff
@@ -261,9 +314,18 @@ def get_nodes_info(nodes="", command_timeout=DEFAULT_GET_INFO_COMMAND_TIMEOUT):
     return _parse_nodes_info(nodeinfo_str)
 
 
-def get_partition_info(command_timeout=DEFAULT_GET_INFO_COMMAND_TIMEOUT, get_all_nodes=True):
-    """Retrieve slurm partition info from scontrol."""
-    show_partition_info_command = f'{SCONTROL} show partitions | grep -oP "^PartitionName=\\K(\\S+)| State=\\K(\\S+)"'
+def get_partition_info(command_timeout=DEFAULT_GET_INFO_COMMAND_TIMEOUT):
+    """
+    Retrieve slurm partition info from scontrol.
+
+    This function considers only partitions managed by ParallelCluster.
+    """
+    partitions = list(PartitionNodelistMapping.instance().get_partitions())
+    grep_filter = _get_partition_grep_filter(partitions)
+    show_partition_info_command = (
```

Comment:
It makes little sense to test this if we have to mock the result of the actual Slurm command.

The `PartitionNodelistMapping.get_partitions()` part is covered by the new test class `TestPartitionNodelistMapping` tests.

Another unit test was created for the `get_partition_grep_filter` function.

The third line is a simple Python f-string.